rfp-proposal/on-chain-quadratic-funding.md

```diff
@@ -0,0 +1,67 @@
+
+### Milestone 1 - Anti-collusion
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | Inline documentation of the code and a basic tutorial that explains how a developer can use the project |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | Article or tutorial that explains the work done as part of the grant. 
+| 1. | Anti-collusion | Implement a mechanism to prevent bribery and collusion, leveraging encrypting votes (ZKPs) potentially via [Minimum Anti-Collusion Infrastructure (MACI)](https://ethresear.ch/t/minimal-anti-collusion-infrastructure/5413) |  
+| 2.  | Voting Example | Integrate a basic voting example that lets you test the mechanism | 
+
+
+### Milestone 2 - Sybil resistance
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | Inline documentation of the code and a basic tutorial that explains how a developer can use the project|
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | Article or tutorial that explains the work done as part of the grant. 
+| 1. | Sybil resistance | Mechanism to prevent Sybil attacks, either implemented as its own Substrate pallet or possible integration of the existing on-chain identity on Kusama/Polkadot provided by specified registrar  |  
+| 2.  | UI | UI for verification of identities, including a polkadot.js extension support | 
+
+### Milestone 3 - CLR
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | Inline documentation of the code and a basic tutorial that explains how a developer can use the project |
+| 0c. | Testing Guide | The code will have unit-test coverage to ensure functionality and robustness. In the guide we will describe how to run these tests | 
+| 0d. | Article/Tutorial | Article or tutorial that explains the work done as part of the grant. 
+| 1. | CLR pallet(s) | Implement the necessary substrate pallets, potentially containing a Token Curated Registry (TCR) to allow anyone to permissionlessly register eligible recipients each round and DAO to govern the protocol.  |  
```

Comment:
Just thinking out loud but can we hook this into democracy somehow? Maybe a new Collective: "CLR funders", into which users with Identities can enter themselves with a deposit (Sybil resistance). The "CLR funders" origin can then execute calls on some CLR pallet functions following the typical AQB or even CLR method of applying votes. In short, tie this into democracy and collective for a demo of cross-pallet integration and reuse.

Reply:
Definitely sounds interesting and whoever is going to implement this, can come with their own ideas on how exactly this should work.